docs/source/basic-usage.rst

```diff
@@ -181,69 +191,50 @@ Closing
 WebSockets are closed with a handshake that requires each endpoint to
 send one frame and receive one frame. Sending a
 :class:`CloseConnection <wsproto.events.CloseConnection>` instance
-places a close frame in the send buffer. When a close frame is
-received, it yields a ``CloseConnection`` event, *and it also places a
-reply frame in the send buffer.* When that reply has been received by
-the initiator, it will also receive a ``CloseConnection`` event.
+sets the state to ``LOCAL_CLOSING``. When a close frame is received,
+it yields a ``CloseConnection`` event, sets the state to
+``REMOTE_CLOSING`` **and requires a reply to be sent**, this reply
+should be a ``CloseConnection`` event. To aid with this the
+``CloseConnection`` class has a :func:`response()
+<wsproto.events.CloseConnection.response>` method to create the
+appropriate reply. For example,
+
+.. code-block:: python
+
+    if isinstance(event, CloseConnection):
+        stream.send(ws.send(event.response()))
```

Comment:
Just a thought: rename `stream` in all docs & examples to `sock`, because at least for me, `sock` is a stronger hint that we are dealing with a OS network socket here. `stream` is a bit undefined, and might even be misleading if people worked with H2 streams previously...

Reply:
Done, throughout.